napari/plugins/pypi.py

```diff
@@ -0,0 +1,63 @@
+"""
+These convenience functions will be useful for searching pypi for packages
+that match the plugin naming convention, and retrieving related metadata.
+"""
```

Comment:
Here also, if we start supporting pyproject.toml, it might be possible to search PyPI, which is (I believe?) not possible with the setup.py entrypoints mechanism...?

Reply:
here too, I am confused about pyproject.toml and what actually gets packaged.  I definitely don't know how to search pyproject metadata at pypi.  However, to search pypi for plugins that are _not_ named by naming convention... I believe the "designated" way is to [request a new Trove Classifier ](https://github.com/pypa/pypi-support/issues/new?labels=classifier+request&template=new-trove-classifier.md&title=Trove+Classifier+Request) from pypa.  And then plugins would add "`Framework :: napari`" to their classifiers in setup.py.  Sounds like they're pretty protective of those classifiers though, so we need to demonstrate a vibrant ecosystem first.

Reply:
That would be pretty nifty! I'm kinda shocked that it seems impossible to associate arbitrary declarative metadata to a package. It looks like "keywords" make it to the package metadata in site-packages, but I'm not sure we want to go that route...

Maybe we need to submit a new PEP for arbitrary metadata...